x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -495,6 +512,60 @@ public boolean maybeFetchFullEntry(KeyType cacheKey, long length, RangeMissingHa
         return true;
     }
 
+    /**
+     * Fetch and write in cache a region of a blob if there are enough free pages in the cache to do so.
+     *
+     * This method returns as soon as the download tasks are instantiated, but the tasks themselves
+     * are run on the bulk executor.
+     *
+     * If an exception is thrown from the writer then the cache entry being downloaded is freed
+     * and unlinked
+     *
+     * @param cacheKey  the key to fetch data for
+     * @param region    the region of the blob to fetch
+     * @param blobLength the length of the blob from which the region is fetched (used to compute the size of the ending region)
+     * @param writer    a writer that handles writing of newly downloaded data to the shared cache
+     * @param listener  a listener that is completed with {@code true} if the current thread triggered the fetching of the region. The
```

Comment:
Can we add that when it is triggered with true, the data is already available.

Reply:
Done in 85c059287230a5c1f746d70b563048b7930a5e3d